tests/compile-fail/never_say_never.rs

```diff
@@ -4,7 +4,7 @@
 fn main() {
     let y = &5;
     let x: ! = unsafe {
-        *(y as *const _ as *const !) //~ ERROR entered unreachable code
+        *(y as *const _ as *const !) //~ ERROR tried to access a dead local variable
```

Comment:
Reads and writes are both accesses, yes. I did not want to introduce separate elements in miri's already way-too-big error type to differentiate read and write...

The rules I implemented for Storage* some weeks ago are that *if* the method uses Storage* at all, then it has to use it for all variables. That's probably stricter than what LLVM requires, but it matches rustc, and it seemed simpler. I think it's quite neat that this manages to catch the `!` here :D